### PR TITLE
feat: self validation of built payload

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Introduce new CLI argument `--debug.payload-building-validation` that enables forwarding built payloads back to consensus engine for validation.

### CODE REVIEW

**Review by Rjected — APPROVED** [review]
lgtm, looks useful

File: crates/node/builder/src/launch/engine.rs [hunk 1]
```diff
@@ -344,15 +349,52 @@ where
...
                     payload = built_payloads.select_next_some() => {
                         if let Some(executed_block) = payload.executed_block() {
-                            debug!(target: "reth::cli", block=?executed_block.block().num_hash(),  "inserting built payload");
-                            eth_service.orchestrator_mut().handler_mut().handler_mut().on_event(EngineApiRequest::InsertExecutedBlock(executed_block).into());
+                            let request = if payload_building_validation_enabled {
+                                debug!(target: "reth::cli", block=?executed_block.block().num_hash(), "inserting built payload with validation");
+                                let (tx, rx) = oneshot::channel();
+                                new_payload_responses.push(rx);
+
+                                let message = BeaconEngineMessage::NewPayload {
```
Comment (mattsse) [inline crates/node/builder/src/launch/engine.rs:362]: not a fan of this massive select arm, this blows up the entire thing and is hard to parse, this loop was intended to be easy to understand.

this feeds data from one select arm into another, which means this can be converted into a helper type instead

File: crates/node/core/src/args/debug.rs [hunk 2]
```diff
@@ -26,6 +26,11 @@ pub struct DebugArgs {
     #[arg(long = "debug.max-block", help_heading = "Debug")]
     pub max_block: Option<u64>,
 
+    /// Flag indicating whether the node should send locally-produced payload
+    /// back to consensus engine for validation.
+    #[arg(long = "debug.payload-building-validation", help_heading = "Debug")]
+    pub payload_building_validation: bool,
```
Comment (mattsse) [inline crates/node/core/src/args/debug.rs:32]: opt-in makes sense for eth, but for op this should always be enabled

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/node/core/src/args/debug.rs (hunk 2 above)
Comment (rkrasiuk) [reply crates/node/core/src/args/debug.rs:32]: Forwarding payload is always enabled, this flag only controls whether the payload will be validated or not

File: crates/node/builder/src/launch/engine_task.rs [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+};
+use tokio::sync::oneshot;
+
+#[must_use = "futures do nothing unless you `.await` or poll them"]
+pub(crate) struct EngineServiceTask<N, Client, E, Network>
+where
+    N: EngineNodeTypes,
+    Client: BlockClient + 'static,
+    E: BlockExecutorProvider + 'static,
+{
```
Comment (mattsse) [inline crates/node/builder/src/launch/engine_task.rs:40]: can we document this a bit, that this is basically the node's main loop

File: crates/node/builder/src/launch/engine_task.rs [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+            built_payload_stream,
+            service,
+            event_sender,
+            initial_backfill_target,
+            payload_validation_enabled,
+            new_payload_responses: FuturesUnordered::default(),
+        }
+    }
+
+    fn on_chain_event(&self, event: ChainEvent<BeaconConsensusEngineEvent>) -> eyre::Result<()> {
```
Comment (mattsse) [inline crates/node/builder/src/launch/engine_task.rs:80]: I would also like some docs here that this is for reporting

File: crates/node/builder/src/launch/engine_task.rs [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+                    };
+                    self.network_handle.update_status(head_block);
+                }
+                self.event_sender.notify(ev);
+            }
+        }
+        Ok(())
+    }
+
+    fn on_built_executed_block(&mut self, executed_block: ExecutedBlock) {
```
Comment (mattsse) [inline crates/node/builder/src/launch/engine_task.rs:113]: docs :) and what this does

File: crates/node/builder/src/launch/engine_task.rs [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+                        versioned_hashes: executed_block
+                            .block
+                            .blob_versioned_hashes_iter()
+                            .copied()
+                            .collect(),
+                    },
+                ),
+                tx,
+            };
+            EngineApiRequest::Beacon(message)
```
Comment (mattsse) [inline crates/node/builder/src/launch/engine_task.rs:133]: this is the only thing that changed right? 
we add it as a different request variant and log the response

imo we could make this a lot simpler if we do this with the main version and simply spawn a new task that awaits this response and logs it, because I really prefer the loop select! for this

**Review by mattsse — CHANGES_REQUESTED** [review]
suggesting to solve this slightly differently

**Comment by mattsse** [comment]
we just removed the Optional executedblock arg from the ethpayload because not useful.

so we can close this imo
